### ISSUE TITLE
type not generic when running with Python < 3.9

### ISSUE THREAD

**Issue by srittau** [python/mypy#10303]
**Bug Report**

Please consider the following stub file:

```python
x: dict[str, str]
y: type[str]
```

When checking it with mypy 0.820+dev.1e96f1df48a6e06ba598fa8230b96671b206cba0 and Python 3.9.4, it passes. Using Python 3.8.9 on the other hand, produces the following output:

```
foo.pyi:2: error: "type" expects no type arguments, but 1 given
```

It's weird that generic `dict` works, but generic `type` doesn't. (Although I'd expect both to work with any Python version.)

This came up during python/typeshed#5192.

**Comment by srittau** [comment]
@JukkaL This doesn't seem to fixes in mypy 0.930:

```
$ mypy --version
mypy 0.930
$ cat foo.pyi 
x: dict[str, str]
y: type[str]
$ mypy --python-version=3.9 foo.pyi 
Success: no issues found in 1 source file
$ mypy --python-version=3.8 foo.pyi 
foo.pyi:2: error: "type" expects no type arguments, but 1 given
Found 1 error in 1 file (checked 1 source file)
```

(I tested in with both Python 3.9.6 and 3.6.14.)

**Comment by AlexWaygood** [comment]
> @JukkaL This doesn't seem to fixed in mypy 0.930:

Interestingly, it appears to be fixed for `.py` files, but _not_ for `.pyi` files. This, in a `.py` file, will pass using Python 3.7:

```python
from __future__ import annotations
foo: type[str]
```

But this, in a `.pyi` file, fails:

```python
foo: type[str]
```

Many similar errors can be found in the CI output for https://github.com/python/typeshed/pull/6681.

**Comment by AlexWaygood** [comment]
Cc. @sobolevn

**Comment by sobolevn** [comment]
Sorry, I cannot not reopen this, but here's my PR: https://github.com/python/mypy/pull/11863

**Comment by hauntsaninja** [comment]
Thanks to sobolevn this should be fixed for pyi with old Pythons!
